device/iot-device-samples/android-sample/app/proguard-rules.pro

```diff
@@ -15,3 +15,26 @@
 #-keepclassmembers class fqcn.of.javascript.interface.for.webview {
 #   public *;
 #}
+
+# JSON parser classes don't mix well with minification because JSON parser classes use reflection
+# based on the name of the field to create the JSON, and minification changes those names. As a result
+# all the message payload serializing and deserializing breaks. These lines explicitly exclude
+# the packages where we keep our JSON parsing classes from minification to avoid this issue
+-keep class com.microsoft.azure.sdk.iot.deps.serializer.* { *; }
+-keep class com.microsoft.azure.sdk.iot.deps.twin.* { *; }
+-keep class com.microsoft.azure.sdk.iot.device.edge.MethodRequest { *; }
+-keep class com.microsoft.azure.sdk.iot.device.hsm.parser.ErrorResponse { *; }
+-keep class com.microsoft.azure.sdk.iot.device.hsm.parser.SignRequest { *; }
+
+# These are the recommended exclusions when using the IoT Hub service client SDK. They are commented
+# out because this sample only uses the IoT Hub device client SDK
+#-keep class com.microsoft.azure.sdk.iot.service.ImportMode { *; }
+#-keep class com.microsoft.azure.sdk.iot.service.AuthenticationMechanism { *; }
+
+# These are the recommended exclusions when using the Device Provisioning Service device client SDK.
+# They are commented out because this sample only uses the IoT Hub device client SDK
+#-keep class com.microsoft.azure.sdk.iot.provisioning.device.internal.parser.* { *; }
+
+# These are the recommended exclusions when using the Device Provisioning Service service client SDK.
+# They are commented out because this sample only uses the IoT Hub device client SDK
+#-keep class com.microsoft.azure.sdk.iot.provisioning.service.configs.* { *; }
```

Comment:
These "-keep class x.y.z" lines just mean that proguard should deliberately **not** minify the class or package in question